Introduction.Rmd

```diff
@@ -137,64 +159,97 @@ Make sure you have a recent version of RStudio. You can check you have the right
+You'll also need a C compiler and a few other command line tools.
+
+If you're on Windows or Mac and you don't already have them, RStudio will install them for you. TO CHECK: is this really true? I've added instructions, for now.
+
+### Windows
 
-* On Windows, download and install [Rtools](http://cran.r-project.org/bin/windows/Rtools/). 
-  NB: this is not an R package!
+On Windows the compiler collection needed for building packages from source
+is called Rtools.
 
-* On Mac, make sure you have either XCode (available for free in the App Store)
-  or the ["Command Line Tools for Xcode"](http://developer.apple.com/downloads).
-  You'll need to have a (free) Apple ID.
+Rtools is **NOT an R package**, so it is not installed with
+`install.packages()`. Instead download it from
+<https://cran.r-project.org/bin/windows/Rtools/> and run the installer.
 
-* On Linux, make sure you've installed not only R, but also the R development
-  tools. For example, on Ubuntu (and Debian) you need to install the 
-  `r-base-dev` package.
+During the Rtools installation you may see a window asking you to "Select
+Additional Tasks".
+
+- Do _not_ select the box for "Edit the system PATH". devtools and RStudio should put Rtools on the `PATH` automatically when it is needed.
+- Do select the box for "Save version information to registry" (it should be selected by default).
+
+### macOS
+
+You need to install the Xcode Command Line Tools, which may already be installed. You can check if they are by running
+
+```{r, eval = FALSE}
+devtools::has_devel()
+```
+
+If they are not installed you have a few options.
+
+- Minimalist approach (recommended): Install Xcode Command Line Tools.
+  In the shell:
+  ```shell
+  xcode-select --install
+  ```
+- Install the current release of full [Xcode from the Mac App Store](https://itunes.apple.com/ca/app/xcode/id497799835?mt=12). This includes a very great deal that you do not need, but it offers the advantage of App Store convenience.
+
+### Linux
+
+Make sure you've installed not only R, but also the R development tools. For example, on Ubuntu (and Debian) you need to install the `r-base-dev` package.
+
+### Verify system prep
 
 You can check that you have everything installed and working by running the following code:
 
 ```{r, eval = FALSE}
 library(devtools)
 has_devel()
-#> '/Library/Frameworks/R.framework/Resources/bin/R' --vanilla CMD SHLIB foo.c 
-#> 
-#> clang -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG 
-#>   -I/usr/local/include -I/usr/local/include/freetype2 -I/opt/X11/include
-#>   -fPIC  -Wall -mtune=core2 -g -O2  -c foo.c -o foo.o
-#> clang -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup
-#>   -single_module -multiply_defined suppress -L/usr/local/lib -o foo.so foo.o 
-#>   -F/Library/Frameworks/R.framework/.. -framework R -Wl,-framework 
-#>   -Wl,CoreFoundation
-[1] TRUE
+#> [1] TRUE
```

Comment:
@jimhester I seem to have a version of `devtools::has_devel()` that produces no visible output 🤔

Reply:
I pulled on this thread.

1. So `has_devel()` is really `pkgbuild::has_devel()`.
2. People have already wondered why it is silent: https://github.com/r-lib/pkgbuild/issues/73
3. Conclusion is that some sort of "package development sitrep" function needs to appear in devtools: https://github.com/r-lib/devtools/issues/1970

Reply:
There is https://github.com/r-lib/pkgbuild/issues/69 related to this as well.